### PR TITLE
Add --custom-data to VM and VMSS create

### LINKED ISSUE #1382: [vm] No option to specify custom data while creating a VM
The 'customData' property in the [create vm API](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines?redirectedfrom=MSDN) is required to provide configuration to [cloud-init](https://help.ubuntu.com/community/CloudInit), the provisioning agent on Ubuntu and an increasing number of Linux distro's.

az vm create doesn't allow me to specify custom data (nor is there any mention of customData in this repo)

### PR DESCRIPTION
**Pull request by devigned**
- add --custom-data to VM and VMSS create
- add help example for --custom-data
- add tests for both VM and VMSS --custom-data

- fix ./scripts/dev_setup.py to skip over empty folders and not fail

fixes #1382

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -190,6 +190,7 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
     register_cli_argument(scope, 'admin_username', help='Username for the VM.', arg_group='Authentication')
     register_cli_argument(scope, 'admin_password', help="Password for the VM if authentication type is 'Password'.", arg_group='Authentication')
     register_cli_argument(scope, 'ssh_key_value', help='SSH public key or public key file path.', completer=FilesCompleter(), type=file_type, arg_group='Authentication')
+    register_cli_argument(scope, 'custom_data', help='Custom init script file or text (cloud-init, cloud-config, etc..)', completer=FilesCompleter(), type=file_type)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Does this truly only support a custom init script? If so, we should consider breaking with Xplat and giving it a more self-documenting name like '--init-script'.

Comment (devigned) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: I like that idea. @twitchax what say you?

Comment (OddBloke) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: user-data is what this is called on other clouds; I would suggest using that verbiage if you're going to change it.

Comment (devigned) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: I think the reason behind calling it custom-data is that it acts differently on different OS's. On Linux distros waagent attepts to execute it, but on Windows, it's just a file I believe (https://blogs.msdn.microsoft.com/kristol/2016/05/16/customdata/).

Comment (OddBloke) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: It looks like other non-cloud-init distros will also store it in a file named CustomData: https://docs.microsoft.com/en-us/azure/virtual-machines/virtual-machines-windows-classic-inject-custom-data

Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: In that case I'm fine with custom-data. 'user-data' is just as vague but breaks with Xplat so might as well stay the same.

Comment (devigned) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: ok, custom-data it is

**Pull request merged**